### PR TITLE
Allow VM setup for MCP

### PR DESCRIPTION
**Pull request by jasonwzm**
The requirements are:
- Cluster should be registered to a fleet
- GCE IdentityProvider should be registered
- Service mesh feature should be enabled

### CODE REVIEW

File: asmcli/commands/validate.sh [hunk 1]
```diff
@@ -37,17 +38,23 @@ validate_dependencies() {
...
     if can_modify_gcp_components; then
       enable_workload_identity
       enable_stackdriver_kubernetes
+      if should_enable_service_mesh_feature; then
+        enable_service_mesh_feature
+      fi
     else
       exit_if_no_workload_identity
       exit_if_stackdriver_not_enabled
+      if should_enable_service_mesh_feature; then
```
Comment (zerobfd) [inline asmcli/commands/validate.sh:55]: Let's not use `should_` here. We need a new function `needs_servicemesh_feature` like `needs_asm`. The fact that we require something is separate from whether we should enable it or not.

**Review by zerobfd — CHANGES_REQUESTED** [review]

File: asmcli/commands/validate.sh (hunk 1 above)
Comment (jasonwzm) [reply asmcli/commands/validate.sh]: Done

File: asmcli/cloudbuild-main.yaml [hunk 2]
```diff
@@ -170,6 +170,44 @@ steps:
...
+      --BUILD_ID "${BUILD_ID}"
+    env:
+    - '_LT_CLUSTER_NAME=long-term-test-cluster-environ'
+    - 'SERVICE_ACCOUNT=${_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com'
+    - 'KEY_FILE=${_KEY_FILE}'
+    - '_CI_ASM_IMAGE_LOCATION=${_IMAGE_LOCATION}'
+    - '_CI_ASM_PKG_LOCATION=${_PKG_LOCATION}'
+    - '_CI_ASM_KPT_BRANCH=${_ASM_PKG_BRANCH}'
+    timeout: 600s
+
```
[the remark this answers is not in the document]
Comment (jasonwzm) [reply asmcli/cloudbuild-main.yaml]: Removed and added unit test.

**Review by zerobfd — APPROVED** [review]

**Pull request merged**